u_fingerprinter.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2017 Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tls
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"golang.org/x/crypto/cryptobyte"
+)
+
+// Fingerprinter is a struct largely for holding options for the FingerprintClientHello func
+type Fingerprinter struct {
+	// KeepPSK will ensure that PSK extensions are passed along into the resulting ClientHelloSpec
```

Comment:
Still could be slightly unclear. Maybe
// KeepPSK will ensure that PSK extensions are passed along into the resulting ClientHelloSpec
// Default(false): Fingerprinter will remove the PSK from the ClientHelloSpec

Reply:
The current implementation returns an error if `KeepPSK` is false and it encounters an `extensionPreSharedKey`. Do you think it's preferable to just ignore it?

Reply:
I think so, yes.